governance/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,164 @@
+
+> [!TIP]
+> Guidelines for writing a good analysis, and what the reviewer will look for:
+>
+> - Show your work: Include steps and tools for each conclusion.
+> - Completeness of risks considered.
+> - Include both implementation and operational failure modes
+> - Provide references to support the reviewer.
+> - The size of the document will likely be proportional to the project's complexity.
+> - The ultimate goal of this document is to identify action items to improve the security of the project. The FMA review process can be accelerated by proactively identifying action items during the writing process.
+
+## Introduction
+
+This document covers implementation of Pectra features in the OP stack. This requires a hard fork due to new transaction types like EIP-7702 transactions. This does not cover generic hardfork related security concerns, which will be addressed in a separate document.
+
+Included EIPs:
+
+-   EIP-7702: Set code transaction
+-   EIP-2537: BLS12-381 precompiles
+-   EIP-2935: Block hashes contract predeploy
+-   EIP-7251: Increase the MAX_EFFECTIVE_BALANCE
+-   EIP-7623: Increase calldata cost
+-   EIP-7002: Execution layer triggerable withdrawals
+-   EIP-6110: Supply validator deposits on chain
+-   EIP-7685: General purpose execution layer requests
+
+## Failure Modes and Recovery Paths
+
+### FM1: Chain halt due to upgrade transaction failing
+
+- **Description:** The chain could halt if a network upgrade transaction can't be included in the upgrade block.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. end-to-end tests created for network upgrade transactions
+  2. same upgrade path as previous predeploys
+- **Detection:** Monitors should trigger an alert immediately
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing the network upgrade transaction and updates to `op-node`.
+
+### FM2: Chain halt due to request hash mismatch
+
+- **Description:** Requests hash was added to the block header. If it mismatches between geth and op-node, the chain could halt
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. End-to-end tests
+  2. Simple validation rules - always must be an empty hash
+- **Detection:** Monitors should trigger an alert immediately.
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing the network upgrade transaction and updates to `op-node`.
+
+### FM3: EIP-7702 transactions cannot be included on the chain
+
+- **Description:** SetCode (EIP-7702) transactions can't be included on chain.
+- **Risk Assessment:** Low impact, low likelihood
+- **Mitigations:** 
+	1. End-to-end tests of set code transactions before and after the fork
+- **Detection:** Sending set code transactions would fail; probably manual detection is most likely.
+- **Recovery Path(s)**: This would require another update and fork.
```

Comment:
Is this detailing a bug where the specification logic isn't sufficient for EIP-7702 txs to land on chain?

I could also see a small RPC bug or something non-logic related to cause an EIP-7702 tx issue that wouldn't necessarily require a hardfork.

Reply:
Updated this FMA with an explanation of what might happen. This would be a fork caused by a consensus bug in a client not allowing EIP-7702 txs after Isthmus is active.